### ISSUE TITLE
The "Sponsor" role or "Initial Authorizer" need to have the ability to submit- API- Backend Changes

### ISSUE THREAD

**Issue by acollad1** [US-EPA-CAMD/PI-1#5976]
The "Sponsor" role or "Initial Authorizer" need to have the ability to submit. Code in easey-auth-api/src/sign/Sign.service.ts (authenticate method) continues to limit submissions to users with the "Submitter" role.  

 

<<<>>> 

if (!user.roles.includes('Submitter')) {
   throw new EaseyException(
     new Error('This requires the Submitter role'),
     HttpStatus.BAD_REQUEST,
     {
       responseObject: 'This requires the Submitter role',
     },
   );
 }

<<<>>> 

 

This logic needs to be expanded to also allow users with the "Sponsor" role or "Initial Authorizer" role to make submissions. If there is already a zenhub ticket for this issue, then please send it along.

**Comment by hamezene** [comment]
@shahbazkhan1999 has already worked on this with Kyle and is ready for testing.

**Comment by mark-hayward-erg** [comment]
@shahbazkhan1999

This code in Sign.service.ts (authenticate method) needs to be updated to allow users with the Initial Authorizer role to submit as well.

There will be users that only have the Initial Authorizer role in CDX, and these users should be treated as Sponsors in ECMPS (i.e., anytime we check for the Sponsor role to allow an action also needs to check for the Initial Authorizer role).   

```
if (!(user.roles.includes('Submitter') || user.roles.includes('Sponsor')) ) {
      throw new EaseyException(
        new Error('This requires the Sponsor or Submitter role'),
        HttpStatus.BAD_REQUEST,
        {
          responseObject: 'This requires the Sponsor or Submitter role',
        },
      );
    }
```

**Comment by shahbazkhan1999** [comment]
@spetros-do  will do peer review

**Comment by shahbazkhan1999** [comment]
Changes are only required at the front end. Already done with that and the frontend ticket is in peer review.

**Comment by alangmaid** [comment]
Blocked by #5994
